src/components/Carousel/Carousel.tsx

```diff
@@ -1,76 +1,96 @@
-import { useEffect, useRef, useState } from 'react';
-import { Carousel as AntCarousel } from 'antd';
-import { CarouselRef } from 'antd/es/carousel';
-import styles from './carousel.module.css';
+import { FC, ReactNode, useEffect, useRef, useState } from 'react';
+import style from './carousel.module.css';
+import { Carousel } from 'antd';
 import SliderButtons from './SliderButtons';
-import { ChildCard } from '@/types/ChildCard';
+import { ResponsiveSettings } from '../../types/responsive';
+import type { CarouselRef } from 'antd/es/carousel';
 
-const Carousels = ({ children, className, responsive }: ChildCard) => {
-  const ref = useRef<CarouselRef | null>(null);
+interface ICarouselsProps {
+  children: ReactNode;
+  classname: string;
+  sliderClassName: string;
+  countSlideToShow: number;
+  responsive?: ResponsiveSettings[];
+}
+
+const Carousels: FC<ICarouselsProps> = ({
+  children,
+  classname,
+  sliderClassName,
+  countSlideToShow,
+  responsive,
+}) => {
+  const ref = useRef<CarouselRef>(null);
   const [prevClick, setPrevClick] = useState(false);
   const [nextClick, setNextClick] = useState(false);
-
-  const [isLastSlick, setIsLastSlick] = useState(false);
   const [isFirstSlick, setIsFirstSlick] = useState(true);
-  const [currentSlickIndex, setCurrentSlickIndex] = useState(0);
+  const [isLastSlick, setIsLastSlick] = useState(false);
+
+  console.log(children);
 
   const handlePrevClick = () => {
     setPrevClick(true);
     ref.current?.prev();
-    setTimeout(() => setPrevClick(false), 0);
+    setTimeout(() => setPrevClick(false), 0); // Reset click state after 0s
   };
 
   const handleNextClick = () => {
     setNextClick(true);
     ref.current?.next();
-    setTimeout(() => setNextClick(false), 0);
+    setTimeout(() => setNextClick(false), 0); // Reset click state after 0s
   };
-
   useEffect(() => {
-    const allSlicks = document.querySelectorAll(
```

Comment:
let's avoid taking DOM elements without React. Seems like you can have thoose elements with taking them from [ref](https://react.dev/learn/referencing-values-with-refs).